packages/rum-core/src/performance-monitoring/transaction-service.js

```diff
@@ -352,6 +355,44 @@ class TransactionService {
     )
   }
 
+  setSession(tr) {
+    let session = this._config.get('session')
+
+    if (session) {
+      if (typeof session == 'boolean') {
+        tr.session = {
+          id: generateRandomId(16),
+          sequence: 1
+        }
+      } else {
+        if (
+          session.timestamp &&
+          Date.now() - session.timestamp > SESSION_TIMEOUT
+        ) {
+          tr.session = {
+            id: generateRandomId(16),
+            sequence: 1
+          }
+        } else {
+          tr.session = {
+            id: session.id,
```

Comment:
feels unnecessary to assign the same id again. maybe directly update the sequence alone.

Reply:
We're setting the id from local storage on the transaction, this is needed here.